app/controllers/assignments_controller.rb

```diff
@@ -35,14 +35,14 @@ def generate_rotation
     user_ids = params.require :user_ids
     start_user = params.require :starting_user_id
     if end_date.before? start_date
-      flash[:errors] = 'The end date must be after the start date.'
-      redirect_back(fallback_location:
-                    roster_assignments_path(@roster)) and return
+      flash.now[:errors] = t('.end_before_start')
+      @start_date = start_date
```

Comment:
```suggestion
```
If we just change `start_date` => `@start_date` on L33, L47 and L52, we don't have to add these at all